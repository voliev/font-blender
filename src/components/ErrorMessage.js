import React from "react";
import PropTypes from "prop-types";
import "./styles/ErrorMessage.css";

const ErrorMessage = ({ children }) => (
  <div className="ErrorMessage">{children}</div>
);

ErrorMessage.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node)
  ]).isRequired
};

export default ErrorMessage;
