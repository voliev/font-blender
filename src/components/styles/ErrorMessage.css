.ErrorMessage {
  align-items: center;
  background-color: var(--error-background);
  border: 3px solid var(--error-border);
  border-radius: var(--border-radius-lg);
  color: var(--error-text);
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: var(--space-md);
  padding: var(--space-md);
}
