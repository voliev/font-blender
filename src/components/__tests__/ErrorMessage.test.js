import React from "react";
import { shallow } from "enzyme";
import ErrorMessage from "../ErrorMessage";

const setup = (msg = <p>Oops! Something went wrong...</p>) =>
  shallow(<ErrorMessage>{msg}</ErrorMessage>);

it("should render without crashing", () => {
  const wrapper = setup();

  expect(wrapper).toMatchSnapshot();
});

it("should render error message", () => {
  const wrapper = setup();

  expect(wrapper.children().length).toBe(1);
  expect(
    wrapper.childAt(0).matchesElement(<p>Oops! Something went wrong...</p>)
  ).toBeTruthy();
});
