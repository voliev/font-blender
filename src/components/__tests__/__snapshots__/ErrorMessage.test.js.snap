// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div
  className="ErrorMessage"
>
  <p>
    Oops! Something went wrong...
  </p>
</div>
`;
